app/controllers/hq/users_controller.rb

```diff
@@ -6,4 +6,24 @@ def index
     @users = User.all
   end
 
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if @user.update_attributes(user_params)
+      flash[:success] = 'User successfully saved'
+      redirect_to hq_user_url(@user)
+    else
+      render 'edit'
+    end
+  end
+
+private
+
+  def user_params
+    params[:user].permit(:user_name, :email)
```

Comment:
Nik I'm really not sure what you're asking me to do about avoiding the model in the case where a mandatory parameter is missing. Are you suggesting that I don't attempt a save in that case and add manually the error message for the user that a mandatory field is missing? This means that any errors on any other fields won't be highlighted, where it's the model which does the validation (eg email address format). 
In the case that the user enters one mandatory field (eg username) but not, say, email address, how do I get the mandatory field missing error message set against the email field but not the username field?